### PR TITLE
Fix publishing headers containing arrays with pecl ext

### PR DESCRIPTION
**Pull request by Domisys**
@odolbeau @stof Following discussions on this [pull request](https://github.com/swarrot/swarrot/pull/111), here is a new proposal for a fix. As you were recommending, the correction has been made on pecl implementation of publisher.

I close previous PR.

### CODE REVIEW

File: src/Swarrot/Broker/MessagePublisher/PeclPackageMessagePublisher.php [hunk 1]
```diff
@@ -38,14 +38,25 @@ public function publish(Message $message, $key = null)
 
             unset($properties['application_headers']);
         }
-
+
```
Comment (odolbeau) [inline src/Swarrot/Broker/MessagePublisher/PeclPackageMessagePublisher.php]: trailing spaces.

File: src/Swarrot/Broker/MessagePublisher/PeclPackageMessagePublisher.php [hunk 2]
```diff
@@ -38,14 +38,25 @@ public function publish(Message $message, $key = null)
...
         $this->exchange->publish(
             $message->getBody(),
             $key,
             $this->flags,
-            $properties
+            $this->removeArraysFromHeaders($properties)
         );
     }
+    
+    private function removeArraysFromHeaders(array $properties)
```
Comment (odolbeau) [inline src/Swarrot/Broker/MessagePublisher/PeclPackageMessagePublisher.php]: Can you:
* rename this method with something like `sanitizeProperties`?
* move the code regarding `delivery_mode` from the RetryProcessor to this method?

Thanks. :)

File: src/Swarrot/Broker/MessagePublisher/PeclPackageMessagePublisher.php (hunk 1 above)
Comment (Domisys) [inline src/Swarrot/Broker/MessagePublisher/PeclPackageMessagePublisher.php]: Fix 70d2b1e5

File: src/Swarrot/Broker/MessagePublisher/PeclPackageMessagePublisher.php (hunk 2 above)
Comment (Domisys) [inline src/Swarrot/Broker/MessagePublisher/PeclPackageMessagePublisher.php]: Fix b177c185 and 56767e15

**Comment by odolbeau** [comment]
Thanks, seems fine for me. :)
Can you squash your commits please?

@stof ok for you?

**Comment by Domisys** [comment]
Done :)

**Comment by webarchi** [comment]
\o/ https://pecl.php.net/package/amqp/1.6.0 Stable release ! \o/

**Comment by odolbeau** [comment]
Good news. :)
BTW thanks for this PR @Domisys & sorry for the delay!

**Pull request merged**